presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -148,34 +148,39 @@ protected RelationPlan visitTable(Table node, Void context)
         Query namedQuery = analysis.getNamedQuery(node);
         Scope scope = analysis.getScope(node);
 
+        RelationPlan plan;
         if (namedQuery != null) {
             RelationPlan subPlan = process(namedQuery, null);
 
             // Add implicit coercions if view query produces types that don't match the declared output types
             // of the view (e.g., if the underlying tables referenced by the view changed)
             Type[] types = scope.getRelationType().getAllFields().stream().map(Field::getType).toArray(Type[]::new);
             RelationPlan withCoercions = addCoercions(subPlan, types);
-            return new RelationPlan(withCoercions.getRoot(), scope, withCoercions.getFieldMappings());
+
+            plan = new RelationPlan(withCoercions.getRoot(), scope, withCoercions.getFieldMappings());
```

Comment:
This block is for Views and CTEs.
We shouldn't try to apply masks/filters on CTEs.
This works probably because we do `columnMasks.getOrDefault` in `io.prestosql.sql.analyzer.Analysis#getColumnMasks`, but this can also mask (no pun intended) bugs.

Reply:
> We shouldn't try to apply masks/filters on CTEs.

We don't. In that case, the Table node will have no column masks associated with it.

Reply:
That's my understanding.